sdk/spring/spring-cloud-azure-integration-tests/src/test/java/com/azure/spring/cloud/integration/tests/eventhubs/binder/EventHubsBinderConsumeErrorIT.java

```diff
@@ -63,13 +63,19 @@ Consumer<Message<String>> consume() {
                 } catch (InterruptedException e) {
                     e.printStackTrace();
                 }
-                throw new RuntimeException();
+                throw new RuntimeException("Consumption exception test");
             };
         }
 
-        @ServiceActivator(inputChannel = "test-eventhub-message.$Default.errors")
-        void consumeError(Message<?> message) throws InterruptedException {
-            EXCHANGER.exchange("ERROR!");
+        @Bean
+        Consumer<MessageHandlingException> consumeError() {
```

Comment:
Why don't we use ErrorMessage?

Reply:
If using the class type `MessageHandlingException`, the message converter will return the payload of the error message from the front converter, that is why the class type `MessageHandlingException` works.

https://github.com/spring-cloud/spring-cloud-function/blob/55494e1cdb7bdf510c06ba8afc0671e05f6d88d2/spring-cloud-function-context/src/main/java/org/springframework/cloud/function/context/config/SmartCompositeMessageConverter.java#L59

if still using the `ErrorMessage`, the IT works, the behind converter will try to convert the payload to the target class type, then an exception  `MessageHandlingException` was thrown, see more from the below console log:
```console
org.springframework.messaging.converter.MessageConversionException: Could not read JSON: Unrecognized token 'org': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: (String)"org.springframework.messaging.MessageHandlingException: error occurred in message handler [org.springframework.cloud.stream.function.FunctionConfiguration$FunctionToDestinationBinder$1@3b87b610], failedMessage=GenericMessage [payload=byte[32], headers={azure_eventhubs_sequence_number=19, azure_partition_key=623600008, azure_checkpointer=com.azure.spring.messaging.checkpoint.AzureCheckpointer@6d6a2bc6, azure_raw_partition_id=0, x-opt-enqueued-time=2022-10-24T07:16:22.994Z, x-opt-sequence-number=1"[truncated 234 chars]; line: 1, column: 4]
	at org.springframework.messaging.converter.MappingJackson2MessageConverter.convertFromInternal(MappingJackson2MessageConverter.java:237) ~[spring-messaging-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.cloud.stream.converter.ApplicationJsonMessageMarshallingConverter.convertFromInternal(ApplicationJsonMessageMarshallingConverter.java:115) ~[spring-cloud-stream-4.0.0-M5.jar:4.0.0-M5]
	at org.springframework.messaging.converter.AbstractMessageConverter.fromMessage(AbstractMessageConverter.java:185) ~[spring-messaging-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.messaging.converter.AbstractMessageConverter.fromMessage(AbstractMessageConverter.java:176) ~[spring-messaging-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.cloud.function.context.config.SmartCompositeMessageConverter.fromMessage(SmartCompositeMessageConverter.java:63) ~[spring-cloud-function-context-4.0.0-M5.jar:4.0.0-M5]
	at org.springframework.cloud.function.context.catalog.SimpleFunctionRegistry$FunctionInvocationWrapper.convertInputMessageIfNecessary(SimpleFunctionRegistry.java:1295) ~[spring-cloud-function-context-4.0.0-M5.jar:4.0.0-M5]
	at org.springframework.cloud.function.context.catalog.SimpleFunctionRegistry$FunctionInvocationWrapper.convertInputIfNecessary(SimpleFunctionRegistry.java:1062) ~[spring-cloud-function-context-4.0.0-M5.jar:4.0.0-M5]
	at org.springframework.cloud.function.context.catalog.SimpleFunctionRegistry$FunctionInvocationWrapper.doApply(SimpleFunctionRegistry.java:704) ~[spring-cloud-function-context-4.0.0-M5.jar:4.0.0-M5]
	at org.springframework.cloud.function.context.catalog.SimpleFunctionRegistry$FunctionInvocationWrapper.apply(SimpleFunctionRegistry.java:559) ~[spring-cloud-function-context-4.0.0-M5.jar:4.0.0-M5]
	at org.springframework.cloud.function.context.catalog.SimpleFunctionRegistry$FunctionInvocationWrapper.accept(SimpleFunctionRegistry.java:575) ~[spring-cloud-function-context-4.0.0-M5.jar:4.0.0-M5]
	at org.springframework.cloud.stream.binder.AbstractMessageChannelBinder.lambda$subscribeFunctionErrorHandler$9(AbstractMessageChannelBinder.java:720) ~[spring-cloud-stream-4.0.0-M5.jar:4.0.0-M5]
	at org.springframework.integration.dispatcher.BroadcastingDispatcher.invokeHandler(BroadcastingDispatcher.java:222) ~[spring-integration-core-6.0.0-M5.jar:6.0.0-M5]
	at org.springframework.integration.dispatcher.BroadcastingDispatcher.dispatch(BroadcastingDispatcher.java:178) ~[spring-integration-core-6.0.0-M5.jar:6.0.0-M5]
	at org.springframework.integration.channel.AbstractSubscribableChannel.doSend(AbstractSubscribableChannel.java:72) ~[spring-integration-core-6.0.0-M5.jar:6.0.0-M5]
	at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:317) ~[spring-integration-core-6.0.0-M5.jar:6.0.0-M5]
	at org.springframework.integration.channel.AbstractMessageChannel.send(AbstractMessageChannel.java:272) ~[spring-integration-core-6.0.0-M5.jar:6.0.0-M5]
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:187) ~[spring-messaging-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:166) ~[spring-messaging-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.messaging.core.GenericMessagingTemplate.doSend(GenericMessagingTemplate.java:47) ~[spring-messaging-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.messaging.core.AbstractMessageSendingTemplate.send(AbstractMessageSendingTemplate.java:109) ~[spring-messaging-6.0.0-M6.jar:6.0.0-M6]
	at org.springframework.integration.endpoint.MessageProducerSupport.sendErrorMessageIfNecessary(MessageProducerSupport.java:272) ~[spring-integration-core-6.0.0-M5.jar:6.0.0-M5]
	at org.springframework.integration.endpoint.MessageProducerSupport.sendMessage(MessageProducerSupport.java:229) ~[spring-integration-core-6.0.0-M5.jar:6.0.0-M5]
	at com.azure.spring.integration.eventhubs.inbound.EventHubsInboundChannelAdapter.access$000(EventHubsInboundChannelAdapter.java:78) ~[classes/:na]
	at com.azure.spring.integration.eventhubs.inbound.EventHubsInboundChannelAdapter$IntegrationRecordMessageListener.onMessage(EventHubsInboundChannelAdapter.java:246) ~[classes/:na]
	at com.azure.spring.integration.eventhubs.inbound.EventHubsInboundChannelAdapter$IntegrationRecordMessageListener.onMessage(EventHubsInboundChannelAdapter.java:227) ~[classes/:na]
	at com.azure.messaging.eventhubs.EventProcessorClientBuilder$1.processEvent(EventProcessorClientBuilder.java:756) ~[azure-messaging-eventhubs-5.13.1.jar:5.13.1]
	at com.azure.messaging.eventhubs.PartitionPumpManager.processEvent(PartitionPumpManager.java:301) ~[azure-messaging-eventhubs-5.13.1.jar:5.13.1]
	at com.azure.messaging.eventhubs.PartitionPumpManager.processEvents(PartitionPumpManager.java:366) ~[azure-messaging-eventhubs-5.13.1.jar:5.13.1]
	at com.azure.messaging.eventhubs.PartitionPumpManager.lambda$startPartitionPump$2(PartitionPumpManager.java:257) ~[azure-messaging-eventhubs-5.13.1.jar:5.13.1]
	at reactor.core.publisher.LambdaSubscriber.onNext(LambdaSubscriber.java:160) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
	at reactor.core.publisher.FluxPublishOn$PublishOnSubscriber.runAsync(FluxPublishOn.java:440) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
	at reactor.core.publisher.FluxPublishOn$PublishOnSubscriber.run(FluxPublishOn.java:527) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
	at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37) ~[reactor-core-3.5.0-M6.jar:3.5.0-M6]
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]
Caused by: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'org': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: (String)"org.springframework.messaging.MessageHandlingException: error occurred in message handler [org.springframework.cloud.stream.function.FunctionConfiguration$FunctionToDestinationBinder$1@3b87b610], failedMessage=GenericMessage [payload=byte[32], headers={azure_eventhubs_sequence_number=19, azure_partition_key=623600008, azure_checkpointer=com.azure.spring.messaging.checkpoint.AzureCheckpointer@6d6a2bc6, azure_raw_partition_id=0, x-opt-enqueued-time=2022-10-24T07:16:22.994Z, x-opt-sequence-number=1"[truncated 234 chars]; line: 1, column: 4]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:2391) ~[jackson-core-2.13.4.jar:2.13.4]
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:745) ~[jackson-core-2.13.4.jar:2.13.4]
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._reportInvalidToken(ReaderBasedJsonParser.java:2961) ~[jackson-core-2.13.4.jar:2.13.4]
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._handleOddValue(ReaderBasedJsonParser.java:2002) ~[jackson-core-2.13.4.jar:2.13.4]
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:802) ~[jackson-core-2.13.4.jar:2.13.4]
	at com.fasterxml.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:4761) ~[jackson-databind-2.13.4.jar:2.13.4]
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4667) ~[jackson-databind-2.13.4.jar:2.13.4]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3629) ~[jackson-databind-2.13.4.jar:2.13.4]
	at org.springframework.messaging.converter.MappingJackson2MessageConverter.convertFromInternal(MappingJackson2MessageConverter.java:232) ~[spring-messaging-6.0.0-M6.jar:6.0.0-M6]
	... 39 common frames omitted

```